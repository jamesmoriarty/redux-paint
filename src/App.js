import React from "react";
import { connect } from "react-redux";
import { makeStyles } from "@material-ui/core/styles";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import Card from "@material-ui/core/Card";
import Container from "@material-ui/core/Container";
import Canvas from "./components/Canvas";
import UndoButton from "./components/UndoButton";
import RedoButton from "./components/RedoButton";

const useStyles = makeStyles((theme) => ({
  title: {
    flexGrow: 1,
  },
  menuButton: {
    marginRight: theme.spacing(2),
  },
  content: {
    flexGrow: 1,
    height: "100vh",
    overflow: "auto",
  },
  container: {
    paddingTop: theme.spacing(8),
    paddingBottom: theme.spacing(8),
  },
  canvas: {
    height: "70vh",
  },
}));

function App({ lines, dispatch }) {
  const classes = useStyles();

  return (
    <div>
      <AppBar position="relative">
        <Toolbar>
          <Typography variant="h6" className={classes.title}>
            Redux Paint
          </Typography>
          <UndoButton className={classes.menuButton}/>
          <RedoButton className={classes.menuButton}/>
        </Toolbar>
      </AppBar>
      <main className={classes.content}>
        <Container maxWidth="lg" className={classes.container}>
          <Card className={classes.canvas} raised={true}>
            <Canvas />
          </Card>
        </Container>
      </main>
    </div>
  );
}

const mapStateToProps = (state) => ({
  lines: state.lines,
  future: state.future,
});

export default connect(mapStateToProps, null)(App);
