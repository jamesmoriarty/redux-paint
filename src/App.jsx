import React from "react";
import { connect } from "react-redux";
import { makeStyles } from "@material-ui/core/styles";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import Card from "@material-ui/core/Card";
import Container from "@material-ui/core/Container";
import Button from "@material-ui/core/Button";
import UndoIcon from "@material-ui/icons/Undo";
import RedoIcon from "@material-ui/icons/Redo";
import GestureIcon from "@material-ui/icons/Gesture";
import LineIcon from "@material-ui/icons/Remove";
import RectIcon from "@material-ui/icons/CheckBoxOutlineBlank";
import ColorIcon from "@material-ui/icons/Palette";
import ButtonGroup from "@material-ui/core/ButtonGroup";
import Canvas from "./components/Canvas";
import ColorPicker from "./components/ColorPicker";
import { OP_TYPES, ACTION_TYPES } from "./constants";

const useStyles = makeStyles((theme) => ({
  title: {
    flexGrow: 1,
  },
  toolButtonGroup: {
    marginRight: theme.spacing(2),
  },
  content: {
    flexGrow: 1,
    height: "100vh",
    overflow: "auto",
  },
  container: {
    paddingTop: theme.spacing(12),
    paddingBottom: theme.spacing(8),
  },
  canvas: {
    height: "70vh",
  },
}));

function App({ history, future, type, dispatch, color }) {
  const classes = useStyles();

  return (
    <div>
      <AppBar position="relative">
        <Toolbar>
          <Typography variant="h6" className={classes.title}>
            Redux Paint
          </Typography>
          <ButtonGroup className={classes.toolButtonGroup}>
            <ColorPicker>
              <Button variant="contained">
                <ColorIcon style={{ color: color }} />
              </Button>
            </ColorPicker>
          </ButtonGroup>
          <ButtonGroup className={classes.toolButtonGroup}>
            <Button
              variant="contained"
              color={
                type === OP_TYPES.OP_TYPE_GESTURE ? "secondary" : "default"
              }
              onClick={() =>
                dispatch({
                  type: ACTION_TYPES.OP_SET_TYPE,
                  payload: { type: OP_TYPES.OP_TYPE_GESTURE },
                })
              }
            >
              <GestureIcon />
            </Button>
            <Button variant="contained" disabled={true}>
              <LineIcon />
            </Button>
            <Button
              variant="contained"
              color={type === OP_TYPES.OP_TYPE_RECT ? "secondary" : "default"}
              onClick={() =>
                dispatch({
                  type: ACTION_TYPES.OP_SET_TYPE,
                  payload: { type: OP_TYPES.OP_TYPE_RECT },
                })
              }
            >
              <RectIcon />
            </Button>
          </ButtonGroup>
          <ButtonGroup className={classes.toolButtonGroup}>
            <Button
              variant="contained"
              disabled={history.length === 0}
              onClick={() => {
                dispatch({ type: ACTION_TYPES.UNDO, payload: {} });
              }}
            >
              <UndoIcon />
            </Button>
            <Button
              variant="contained"
              disabled={future.length === 0}
              onClick={() => {
                dispatch({ type: ACTION_TYPES.REDO, payload: {} });
              }}
            >
              <RedoIcon />
            </Button>
          </ButtonGroup>
        </Toolbar>
      </AppBar>
      <main className={classes.content}>
        <Container className={classes.container}>
          <Card className={classes.canvas} raised={true}>
            <Canvas />
          </Card>
        </Container>
      </main>
    </div>
  );
}

const mapStateToProps = (state) => state;

export default connect(mapStateToProps, null)(App);
