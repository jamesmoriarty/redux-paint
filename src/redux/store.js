import { configureStore } from "@reduxjs/toolkit";
import reduce from "./opReducer";

const store = configureStore({ reducer: reduce });

// store.subscribe(() => console.log(store.getState()));

export default store;
