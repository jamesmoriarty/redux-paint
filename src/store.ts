import { configureStore } from "@reduxjs/toolkit";
import reduce from "./reducer";

const store = configureStore({ reducer: reduce });

// store.subscribe(() => console.log(store.getState()));

export default store;
