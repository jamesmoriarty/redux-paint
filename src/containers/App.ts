import App from "../App";
import { ACTION_TYPES, OP_TYPES, State, Dispatch } from "../types";
import { connect } from "react-redux";

const mapStateToProps = (state: State) => state;

const mapDispatchToProps = (dispatch: Dispatch) => {
  return {
    onGesture: () => {
      dispatch({
        type: ACTION_TYPES.OP_SET_TYPE,
        payload: { type: OP_TYPES.OP_TYPE_GESTURE },
      });
    },
    onRect: () => {
      dispatch({
        type: ACTION_TYPES.OP_SET_TYPE,
        payload: { type: OP_TYPES.OP_TYPE_RECT },
      });
    },
    onUndo: () => {
      dispatch({ type: ACTION_TYPES.UNDO, payload: {} });
    },
    onRedo: () => {
      dispatch({ type: ACTION_TYPES.REDO, payload: {} });
    },
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(App);
