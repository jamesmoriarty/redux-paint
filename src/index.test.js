import React from "react";
import { act, render } from "@testing-library/react";
import App from "./containers/App";
import { Provider } from "react-redux";
import store from "./store";

test("renders html", () => {
  act(() => {
    expect(
      render(
        <Provider store={store}>
          <App />
        </Provider>
      )
    ).toMatchSnapshot();
  });
});
