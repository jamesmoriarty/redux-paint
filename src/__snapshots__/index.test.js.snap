// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders html 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <header
          class="MuiPaper-root MuiAppBar-root MuiAppBar-positionRelative MuiAppBar-colorPrimary MuiPaper-elevation4"
        >
          <div
            class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
          >
            <p
              class="MuiTypography-root makeStyles-title-1 MuiTypography-body1"
            />
            <div
              class="MuiButtonGroup-root makeStyles-toolButtonGroup-2"
              role="group"
            >
              <div>
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      style="color: rgb(51, 51, 51);"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 3c-4.97 0-9 4.03-9 9s4.03 9 9 9c.83 0 1.5-.67 1.5-1.5 0-.39-.15-.74-.39-1.01-.23-.26-.38-.61-.38-.99 0-.83.67-1.5 1.5-1.5H16c2.76 0 5-2.24 5-5 0-4.42-4.03-8-9-8zm-5.5 9c-.83 0-1.5-.67-1.5-1.5S5.67 9 6.5 9 8 9.67 8 10.5 7.33 12 6.5 12zm3-4C8.67 8 8 7.33 8 6.5S8.67 5 9.5 5s1.5.67 1.5 1.5S10.33 8 9.5 8zm5 0c-.83 0-1.5-.67-1.5-1.5S13.67 5 14.5 5s1.5.67 1.5 1.5S15.33 8 14.5 8zm3 4c-.83 0-1.5-.67-1.5-1.5S16.67 9 17.5 9s1.5.67 1.5 1.5-.67 1.5-1.5 1.5z"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
            <div
              class="MuiButtonGroup-root makeStyles-toolButtonGroup-2"
              role="group"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlined MuiButton-containedSecondary"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M4.59 6.89c.7-.71 1.4-1.35 1.71-1.22.5.2 0 1.03-.3 1.52-.25.42-2.86 3.89-2.86 6.31 0 1.28.48 2.34 1.34 2.98.75.56 1.74.73 2.64.46 1.07-.31 1.95-1.4 3.06-2.77 1.21-1.49 2.83-3.44 4.08-3.44 1.63 0 1.65 1.01 1.76 1.79-3.78.64-5.38 3.67-5.38 5.37 0 1.7 1.44 3.09 3.21 3.09 1.63 0 4.29-1.33 4.69-6.1H21v-2.5h-2.47c-.15-1.65-1.09-4.2-4.03-4.2-2.25 0-4.18 1.91-4.94 2.84-.58.73-2.06 2.48-2.29 2.72-.25.3-.68.84-1.11.84-.45 0-.72-.83-.36-1.92.35-1.09 1.4-2.86 1.85-3.52.78-1.14 1.3-1.92 1.3-3.28C8.95 3.69 7.31 3 6.44 3 5.12 3 3.97 4 3.72 4.25c-.36.36-.66.66-.88.93l1.75 1.71zm9.29 11.66c-.31 0-.74-.26-.74-.72 0-.6.73-2.2 2.87-2.76-.3 2.69-1.43 3.48-2.13 3.48z"
                    />
                  </svg>
                </span>
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlined"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                </span>
              </button>
            </div>
            <div
              class="MuiButtonGroup-root makeStyles-toolButtonGroup-2"
              role="group"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlined Mui-disabled Mui-disabled"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12.5 8c-2.65 0-5.05.99-6.9 2.6L2 7v9h9l-3.62-3.62c1.39-1.16 3.16-1.88 5.12-1.88 3.54 0 6.55 2.31 7.6 5.5l2.37-.78C21.08 11.03 17.15 8 12.5 8z"
                    />
                  </svg>
                </span>
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlined Mui-disabled Mui-disabled"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M18.4 10.6C16.55 8.99 14.15 8 11.5 8c-4.65 0-8.58 3.03-9.96 7.22L3.9 16c1.05-3.19 4.05-5.5 7.6-5.5 1.95 0 3.73.72 5.12 1.88L13 16h9V7l-3.6 3.6z"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </header>
        <main
          class="makeStyles-content-3"
        >
          <div
            class="MuiContainer-root makeStyles-container-4 MuiContainer-maxWidthLg"
          >
            <div
              class="MuiPaper-root MuiCard-root makeStyles-canvas-5 MuiPaper-elevation8 MuiPaper-rounded"
            >
              <canvas />
            </div>
          </div>
        </main>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionRelative MuiAppBar-colorPrimary MuiPaper-elevation4"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
        >
          <p
            class="MuiTypography-root makeStyles-title-1 MuiTypography-body1"
          />
          <div
            class="MuiButtonGroup-root makeStyles-toolButtonGroup-2"
            role="group"
          >
            <div>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    style="color: rgb(51, 51, 51);"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 3c-4.97 0-9 4.03-9 9s4.03 9 9 9c.83 0 1.5-.67 1.5-1.5 0-.39-.15-.74-.39-1.01-.23-.26-.38-.61-.38-.99 0-.83.67-1.5 1.5-1.5H16c2.76 0 5-2.24 5-5 0-4.42-4.03-8-9-8zm-5.5 9c-.83 0-1.5-.67-1.5-1.5S5.67 9 6.5 9 8 9.67 8 10.5 7.33 12 6.5 12zm3-4C8.67 8 8 7.33 8 6.5S8.67 5 9.5 5s1.5.67 1.5 1.5S10.33 8 9.5 8zm5 0c-.83 0-1.5-.67-1.5-1.5S13.67 5 14.5 5s1.5.67 1.5 1.5S15.33 8 14.5 8zm3 4c-.83 0-1.5-.67-1.5-1.5S16.67 9 17.5 9s1.5.67 1.5 1.5-.67 1.5-1.5 1.5z"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </div>
          <div
            class="MuiButtonGroup-root makeStyles-toolButtonGroup-2"
            role="group"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlined MuiButton-containedSecondary"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M4.59 6.89c.7-.71 1.4-1.35 1.71-1.22.5.2 0 1.03-.3 1.52-.25.42-2.86 3.89-2.86 6.31 0 1.28.48 2.34 1.34 2.98.75.56 1.74.73 2.64.46 1.07-.31 1.95-1.4 3.06-2.77 1.21-1.49 2.83-3.44 4.08-3.44 1.63 0 1.65 1.01 1.76 1.79-3.78.64-5.38 3.67-5.38 5.37 0 1.7 1.44 3.09 3.21 3.09 1.63 0 4.29-1.33 4.69-6.1H21v-2.5h-2.47c-.15-1.65-1.09-4.2-4.03-4.2-2.25 0-4.18 1.91-4.94 2.84-.58.73-2.06 2.48-2.29 2.72-.25.3-.68.84-1.11.84-.45 0-.72-.83-.36-1.92.35-1.09 1.4-2.86 1.85-3.52.78-1.14 1.3-1.92 1.3-3.28C8.95 3.69 7.31 3 6.44 3 5.12 3 3.97 4 3.72 4.25c-.36.36-.66.66-.88.93l1.75 1.71zm9.29 11.66c-.31 0-.74-.26-.74-.72 0-.6.73-2.2 2.87-2.76-.3 2.69-1.43 3.48-2.13 3.48z"
                  />
                </svg>
              </span>
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlined"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </span>
            </button>
          </div>
          <div
            class="MuiButtonGroup-root makeStyles-toolButtonGroup-2"
            role="group"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlined Mui-disabled Mui-disabled"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12.5 8c-2.65 0-5.05.99-6.9 2.6L2 7v9h9l-3.62-3.62c1.39-1.16 3.16-1.88 5.12-1.88 3.54 0 6.55 2.31 7.6 5.5l2.37-.78C21.08 11.03 17.15 8 12.5 8z"
                  />
                </svg>
              </span>
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlined Mui-disabled Mui-disabled"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M18.4 10.6C16.55 8.99 14.15 8 11.5 8c-4.65 0-8.58 3.03-9.96 7.22L3.9 16c1.05-3.19 4.05-5.5 7.6-5.5 1.95 0 3.73.72 5.12 1.88L13 16h9V7l-3.6 3.6z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </header>
      <main
        class="makeStyles-content-3"
      >
        <div
          class="MuiContainer-root makeStyles-container-4 MuiContainer-maxWidthLg"
        >
          <div
            class="MuiPaper-root MuiCard-root makeStyles-canvas-5 MuiPaper-elevation8 MuiPaper-rounded"
          >
            <canvas />
          </div>
        </div>
      </main>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
